core/src/dispatch/mod.rs

```diff
@@ -761,6 +848,39 @@ impl NetworkDispatcher {
             RegistrationPromise::None => (), // ignore
         }
     }
+
+    fn close_channel(&mut self, addr: SocketAddr) -> () {
+        if let Some(state) = self.connections.get_mut(&addr) {
+            match state {
+                ConnectionState::Connected(_) => {
+                    debug!(
+                        self.ctx.log(),
+                        "Closing channel to connected system {}", addr
+                    );
+                    if let Some(bridge) = &self.net_bridge {
+                        while self.queue_manager.has_data(&addr) {
+                            if let Some(data) = self.queue_manager.pop_data(&addr) {
+                                if let Err(e) = bridge.route(addr, data, Protocol::TCP) {
+                                    error!(self.ctx.log(), "Bridge error while routing {:?}", e);
+                                }
+                            }
+                        }
+                        if let Err(e) = bridge.close_channel(addr) {
+                            error!(self.ctx.log(), "Bridge error closing channel {:?}", e);
+                        }
+                    }
+                }
+                _ => {
+                    debug!(
+                        self.ctx.log(),
+                        "Trying to close channel to a system which is not connected {}", addr
+                    );
+                }
+            }
+        } else {
+            debug!(self.ctx.log(), "Closing channel to unknown system {}", addr);
```

Comment:
I would make this and the one above warning actually. This could indicate a bug on the side of the requester.